tests/cupy_tests/indexing_tests/test_generate.py

```diff
@@ -24,3 +25,105 @@ def test_ix_empty_ndarray(self, xp):
     @testing.numpy_cupy_array_list_equal()
     def test_ix_bool_ndarray(self, xp):
         return xp.ix_(xp.array([True, False] * 2))
+
+
+@testing.gpu
+class TestR_(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_r_1(self, xp, dtype):
+        a = testing.shaped_arange((3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 4), xp, dtype)
+        return xp.r_[a, b]
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_r_8(self, xp, dtype):
+        a = testing.shaped_arange((3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 4), xp, dtype)
+        c = testing.shaped_reverse_arange((1, 4), xp, dtype)
+        return xp.r_[a, b, c]
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal(type_check=False)
```

Comment:
Yes, but in my environment, this error has occured if type_check=true. Do you know this mean? ( or how to fix.)

======================================================================
FAIL: test_r_2 (test_generate.TestR_)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/www/now1202/chainer/cupy/testing/helper.py", line 404, in test_func
    impl(self, *args, **kw)
  File "/var/www/now1202/chainer/cupy/testing/helper.py", line 117, in test_func
    self.assertEqual(cupy_result.dtype, numpy_result.dtype)
AssertionError: dtype('float64') != dtype('float32')
-------------------- >> begin captured stdout << ---------------------
dtype is <class 'numpy.float32'>

--------------------- >> end captured stdout << ----------------------